src/main/java/graphql/execution/preparsed/persisted/PersistedQuerySupport.java

```diff
@@ -36,7 +37,7 @@ public PersistedQuerySupport(PersistedQueryCache persistedQueryCache) {
     }
 
     @Override
-    public PreparsedDocumentEntry getDocument(ExecutionInput executionInput, Function<ExecutionInput, PreparsedDocumentEntry> parseAndValidateFunction) {
```

Comment:
Same pattern here.  We delegate back from an async method to this method

This is also `@PublicSpi` and hence people will have derived from it and hence we dont want to break them